component/src/main/java/org/wso2/extension/siddhi/execution/string/FillTemplateFunctionExtension.java

```diff
@@ -0,0 +1,140 @@
+                                "There can be any number of strings from the 2nd argument, as comma separated values.",
+                        type = {DataType.STRING}),
+        },
+        returnAttributes =
+            @ReturnAttribute(
+                description = "String in which templated positions filled with the given strings.",
+                type = DataType.STRING),
+        examples =
+            @Example(
+                description = "" +
+                        "In this example, the templated string with index 1 {{1}} will be replaced with the 1st " +
+                        "string value provided ('an example').\n" +
+                        "{{2}} will be replaced with the 2nd string provided ('fillTemplate')\n" +
+                        "The return string will be 'This is an example for the fillTemplate function'.",
+                syntax = "fillTemplate(\"This is {{1}} for the {{2}} function\",  'an example', 'fillTemplate')")
+)
+public class FillTemplateFunctionExtension extends FunctionExecutor {
+    Pattern templatePattern = Pattern.compile("(\\{\\{\\d}})");
+    Pattern indexPattern = Pattern.compile("\\d");
+
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                        SiddhiAppContext siddhiAppContext) {
+        if (attributeExpressionExecutors.length <= 1) {
+            throw new SiddhiAppValidationException("Invalid number of arguments passed to " +
+                    "str:fillTemplate() function. " +
+                    "Required at least 2, but found " + attributeExpressionExecutors.length);
+        }
+
+        for (int i = 0; i < attributeExpressionExecutors.length; i++) {
+            if (attributeExpressionExecutors[i].getReturnType() != Attribute.Type.STRING) {
+                throw new SiddhiAppValidationException("Invalid parameter type found for the argument no " + (i + 1) +
+                        ". All the parameters should be in type " + Attribute.Type.STRING + ", but found " +
+                        attributeExpressionExecutors[i].getReturnType().toString());
+            }
+        }
+    }
+
+    @Override
+    protected Object execute(Object[] data) {
+        String sourceString = (String) data[0];
+        Matcher templateMatcher = templatePattern.matcher(sourceString);
+        String match;
+        Matcher indexMatcher;
+        int index;
+        while (templateMatcher.find()) {
+            match = templateMatcher.group(0);
+            indexMatcher = indexPattern.matcher(match);
+            if (indexMatcher.find()) {
+                index = Integer.parseInt(indexMatcher.group(0));
+                if (index < 1 || index >= data.length) {
+                    throw new SiddhiAppRuntimeException("Index given for template elements should be greater than 0 " +
+                            "and less than '" + data.length + "'.");
```

Comment:
Say but found what? and also for this exception occurred for which template string